config/300-kafkasource.yaml

```diff
@@ -0,0 +1,399 @@
+                    properties:
+                      sslCA:
+                        description: The value to the configured CA.
+                        type: object
+                        properties:
+                          valueFromSecret:
+                            description: A reference to a Kubernetes Secret object containing the value.
+                            type: object
+                            properties:
+                              name:
+                                type: string
+                              key:
+                                type: string
+                            required:
+                            - name
+                            - key
+                        required:
+                        - valueFromSecret
+                      sslClientCert:
+                        description: The value of the SSL Client Cert.
+                        type: object
+                        properties:
+                          valueFromSecret:
+                            description: A reference to a Kubernetes Secret object containing the value.
+                            type: object
+                            properties:
+                              name:
+                                type: string
+                              key:
+                                type: string
+                            required:
+                            - name
+                            - key
+                        required:
+                        - valueFromSecret
+                      sslClientKey:
+                        description: The value of the SSL Client Key.
+                        type: object
+                        properties:
+                          valueFromSecret:
+                            description: A reference to a Kubernetes Secret object containing the value.
+                            type: object
+                            properties:
+                              name:
+                                type: string
+                              key:
+                                type: string
+                            required:
+                            - name
+                            - key
+                        required:
+                        - valueFromSecret
+                      insecureSkipVerify:
+                        description: InsecureSkipVerify controls whether a client verifies the server's certificate chain
+                          and host name. If InsecureSkipVerify is true, crypto/tls accepts any certificate presented by the
+                          server and any host name in that certificate. In this mode, TLS is susceptible to machine-in-the-middle
+                          attacks unless custom verification is used. This should be used only for testing.
+                        type: boolean
+                required:
+                - saslEnable
+                - tlsEnable
```

Comment:
After reading the CRD, is `tlsEnable` needed or could we just just rely on the tls parameters being informed?

Reply:
This is needed to differentiate between `SASL_PLAIN` and `SASL_SSL` when authenticating (hit this while trying to connect to Oracle's streaming service over the weekend.